shipyard_blueprints/microsoft-onedrive/shipyard_microsoft_onedrive/cli/onedrive_upload.py

```diff
@@ -0,0 +1,117 @@
+
+logger = ShipyardLogger.get_logger()
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--access-token", required=False, help="Access token to use for authentication"
+    )
+    parser.add_argument(
+        "--client-id", required=False, help="Client ID for OAuth authentication"
+    )
+    parser.add_argument(
+        "--client-secret", required=False, help="Client secret for OAuth authentication"
+    )
+    parser.add_argument(
+        "--tenant", required=False, help="Tenant for OAuth authentication"
+    )
+    parser.add_argument("--file-name", required=True, help="Name of the file to upload")
+    parser.add_argument("--directory", default="", required=False)
+    parser.add_argument(
+        "--user-email", required=True, help="Email of the user to upload the file to"
+    )
+    parser.add_argument(
+        "--onedrive-file-name",
+        required=False,
+        default="",
+        help="Name of the file in OneDrive",
+    )
+    parser.add_argument(
+        "--onedrive-directory",
+        required=False,
+        default="",
+        help="Directory in OneDrive to upload the file to",
+    )
+    parser.add_argument(
+        "--match-type",
+        required=False,
+        default="exact_match",
+        choices=["exact_match", "regex_match"],
+        help="Type of matching to use when finding the file in the directory",
+    )
+    return parser.parse_args()
+
+
+def main():
+    try:
+        args = get_args()
+        access_token = args.access_token
+        client_id = args.client_id
+        client_secret = args.client_secret
+        tenant = args.tenant
+        src_file = args.file_name
+        src_dir = args.directory
+        src_path = shipyard.files.combine_folder_and_file_name(src_dir, src_file)
+        user_email = args.user_email
+        target_file = args.onedrive_file_name
+        target_dir = args.onedrive_directory
+
+        target_file = target_file if target_file else src_file
+        target_path = shipyard.files.combine_folder_and_file_name(
+            target_dir, target_file
+        )
+
+        onedrive = None
+        if client_id and client_secret and tenant:
+            onedrive = OneDriveClient(auth_type="basic")
+            onedrive.connect(client_id, client_secret, tenant)
+        else:
+            onedrive = OneDriveClient(auth_type="oauth", access_token=access_token)
+
+        user_id = onedrive.get_user_id(user_email)
+        drive_id = onedrive.get_drive_id(user_id)
+        if target_dir:
+            folder_id = onedrive.get_folder_id(target_dir, drive_id)
+            if not folder_id:
+                onedrive.create_folder(target_dir, drive_id)
+
+        if args.match_type == "exact_match":
+            onedrive.upload(src_path, drive_id, target_path)
+        elif args.match_type == "regex_match":
+            file_names = shipyard.files.find_all_local_file_names(src_dir)
```

Comment:
I just tried to use this and it was improperly naming the new file. Going to leave it as is for now

Reply:
Do you mind opening a bug ticket for the shipyard utils with the examples so that we don't forget to come back to this later. We should fix it so it is done correctly in the future across the board-